src/ussd_pmtct_rapidpro.js

```diff
@@ -1,24 +1,413 @@
+                check: function(content) {
+                    if(!utils.is_valid_msisdn(content, "ZA")) {
+                        return (
+                            "Sorry, we don't understand that cell number. Please enter 10 digit " +
+                            "cell number that the mother would like to get HIV-related messages " +
+                            "on, e.g. 0813547654.");
+                    }
+                    if(utils.normalize_msisdn(content, "ZA") === "+27813547654") {
+                        return (
+                            "We're looking for the mother's information. Please avoid entering " +
+                            "the examples in the messages. Enter the mother's details."
+                        );
+                    }
+                },
+                next: "state_check_subscription"
+            });
+        });
+
+        self.add("state_check_subscription", function(name, opts) {
+            var msisdn = utils.normalize_msisdn(
+                _.get(self.im.user.answers, "state_enter_msisdn", self.im.user.addr), "ZA");
+
+            return self.rapidpro.get_contact({urn: "tel:" + msisdn})
+                .then(function(contact) {
+                    self.im.user.set_answer("contact", contact);
+                    // Set the language if we have it
+                    if(_.isString(_.get(contact, "language"))) {
+                        return self.im.user.set_lang(contact.language);
+                    }
+                }).then(function() {
+                    // Delegate to the correct state depending on group membership
+                    var contact = self.im.user.get_answer("contact");
+                    if(self.contact_in_group(contact, self.im.config.clinic_group_ids)){
+                        if(self.contact_in_group(contact, self.im.config.pmtct_group_ids)){
+                            return self.states.create("state_optout");
+                        } else {
+                            return self.states.create("state_no_pmtct_subscription");
+                        }
+                    } else {
+                        return self.states.create("state_no_subscription");
+                    }
+                }).catch(function(e) {
+                    // Go to error state after 3 failed HTTP requests
+                    opts.http_error_count = _.get(opts, "http_error_count", 0) + 1;
+                    if(opts.http_error_count === 3) {
+                        self.im.log.error(e.message);
+                        return self.states.create("__error__");
+                    }
+                    return self.states.create(name, opts);
+                });
+        });
+
+        self.states.add("state_optout", function(name) {
+            return new ChoiceState(name, {
+                question: "The mother is currently receiving messages about keeping her baby " +
+                            "HIV-negative. Does she want to stop getting these messages?",
```

Comment:
I've noticed that a few of the user strings aren't surrounded by `$()`, which means that we won't be able to translate them. Any of the strings after the `self.im.user.set_lang(contact.language)` needs to be surrounded in `$()`, otherwise they won't be displayed in the user's language.

Reply:
but according to the flow, that only happens after the `no active PMTCT` part of the flow?

Reply:
I think that should be moved up, because we do know their language for the opt out. In the code we're setting their language in `state_check_subscription`, so any state that comes after that must use the translation strings, and any state before that mustn't.

So either we can change where we set the language, or we update the flow to set the language where we're setting it in the code.